app/models/vm_or_template.rb

```diff
@@ -282,6 +280,7 @@ def from_infra_manager?
     virtual_column m, :type => :string, :uses => :all_relationships
   end
 
+  has_ancestry
```

Comment:
Think we need to set `:orphan_strategy => : adopt `

Not sure the exact value here.